sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/UnivocityParser.scala

```diff
@@ -85,13 +85,13 @@ class UnivocityParser(
   // We preallocate it avoid unnecessary allocations.
   private val noRows = None
 
-  private val timestampFormatter = TimestampFormatter(
+  private lazy val timestampFormatter = TimestampFormatter(
```

Comment:
What is the reason to make it lazy?

Reply:
the formatter creation will validate the pattern string now, but json/csv has a fallback and shouldn't fail because of invalid pattern string.